src/main/java/seedu/address/model/student/Student.java

```diff
@@ -27,31 +27,47 @@ public class Student {
 
     // Data fields
     private final Set<Tag> tags = new HashSet<>();
+    private final String remark;
     //AssignmentList initially null
     private List<Assignment> assignmentList = new ArrayList<Assignment>();
 
     /**
      * Every field except assignmentList must be present and not null.
      */
-    public Student(Name name, Phone phone, Email email, Set<Tag> tags) {
-        requireAllNonNull(name, phone, email, tags);
+    public Student(Name name, Phone phone, Email email, Set<Tag> tags, String remark) {
+        requireAllNonNull(name, phone, email, tags, remark);
```

Comment:
Hmm, wondering if we should retain a Student constructor that does not require a remark. 
Does this handle cases where the addCommand works without a remark required? 
We may want to preserve it so that there is no need to assign any remark for new students.

Reply:
Yeah I think that works. Maybe can declare it as a constant too like `public static String EMPTY_REMARK = ''`

Reply:
ok i have made a new constructor and also made a empty remark constant!